tasks/remote_storage.py

```diff
@@ -1,9 +1,221 @@
 import os
 import re
+import concurrent
+import socket
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime
 
-from .common import PopenTask, TaskException
-from .constants import (FEDORAPEOPLE_KEY_PATH, FEDORAPEOPLE_DIR, UUID_RE,
-                        JOBS_DIR)
+import logging
+import boto3
+from botocore.exceptions import ClientError
+from jinja2 import Template
+
+from .common import PopenTask, TaskException, FallibleTask
+from .constants import (CLOUD_JOBS_DIR, CLOUD_JOBS_URL, CLOUD_URL, CLOUD_DIR,
+                        CLOUD_BUCKET, UUID_RE, JOBS_DIR, TASKS_DIR)
+
+
+"""
+Previously we were updating test results in Fedora infra where the results were
+directly served as part of web server directory listing capabilities. This is
+not case of AWS S3.
+In order to simulate same environment we are generating "index.html" using
+Jinja2 template and putting it into every directory. Then we upload particular
+job directory using awscli (which does parallelism for us) under S3 bucket
+"jobs" prefix (directory). Currently 2 awscli commands are needed in order to
+set correct encoding for gzip files. Content types are set automatically using
+"/etc/mime.types" file.
+At the end we create root jobs index to list all "freeipa" repo PR jobs in the
+bucket.
+"""
+
+
+def get_jobdir_metadata(job_uuid):
+    """
+    Job metadata for root index.
+    """
+    try:
+        client = boto3.client('s3')
+        job = client.head_object(Bucket=CLOUD_BUCKET, Key=job_uuid)
+        name = job_uuid.split(os.sep)[-2]
+        if re.match(UUID_RE, name):
+            mtime = job.get('LastModified', None)
+            metadata = job.get('Metadata', dict())
+            repo_owner = metadata.get('repo_owner', None)
+            pr_number = metadata.get('pr_number', None)
+            pr_author = metadata.get('pr_author', None)
+            task_name = metadata.get('task_name', None)
+            returncode = metadata.get('returncode', None)
```

Comment:
Hello, dictionary get method will give you `None` anyway :) https://docs.python.org/3/library/stdtypes.html#dict.get

Reply:
Yep, good eye...fixed